### PR TITLE
debug: add OpenGL enhanced debugging

### PR DESCRIPTION
**Pull request by jmoguillansky-gpsw**

### CODE REVIEW

File: libnodegl/features.h [hunk 1]
```diff
@@ -53,6 +53,7 @@
 #define NGLI_FEATURE_SOFTWARE                     (1 << 28)
 #define NGLI_FEATURE_UINT_UNIFORMS                (1 << 29)
 #define NGLI_FEATURE_EGL_ANDROID_GET_IMAGE_NATIVE_CLIENT_BUFFER (1 << 30)
+#define NGLI_FEATURE_GL_KHR_DEBUG                 (1ULL << 32)
```
Comment (mbouron-gpsw) [inline libnodegl/features.h:56]: Should be (1ULL << 31).

File: libnodegl/gctx_gl.c [hunk 2]
```diff
@@ -220,6 +220,24 @@ static struct gctx *gl_create(const struct ngl_config *config)
...
+#ifdef ENABLE_DEBUG
+#define GL_DEBUG_LOG(log_level, ...) ngli_log_print(log_level, __FILE__, __LINE__, __FUNCTION__, __VA_ARGS__)
+
+static void debug_message_callback( GLenum source,
+                 GLenum type,
+                 GLuint id,
+                 GLenum severity,
+                 GLsizei length,
+                 const GLchar* message,
+                 const void* userParam )
```
Comment (mbouron-gpsw) [inline libnodegl/gctx_gl.c:232]: style: 
* space before first argument,
* align all arguments vertically
* GLchar* message -> GLChar *message
* const void* userParam -> const void *user_param
* no space after last argument

File: libnodegl/gctx_gl.c [hunk 3]
```diff
@@ -220,6 +220,24 @@ static struct gctx *gl_create(const struct ngl_config *config)
...
+{
+    if (severity == GL_DEBUG_SEVERITY_NOTIFICATION) return;
```
Comment (mbouron-gpsw) [inline libnodegl/gctx_gl.c:234]: return statement should be placed on a new line

File: libnodegl/gctx_gl.c [hunk 4]
```diff
@@ -220,6 +220,24 @@ static struct gctx *gl_create(const struct ngl_config *config)
     return (struct gctx *)s;
 }
 
+#ifdef ENABLE_DEBUG
+#define GL_DEBUG_LOG(log_level, ...) ngli_log_print(log_level, __FILE__, __LINE__, __FUNCTION__, __VA_ARGS__)
+
+static void debug_message_callback( GLenum source,
```
Comment (mbouron-gpsw) [inline libnodegl/gctx_gl.c:226]: Please rename it to gl_debug_callback

File: libnodegl/glcontext_egl.c [hunk 5]
```diff
@@ -318,6 +318,9 @@ try_again:;
                 EGL_CONTEXT_MAJOR_VERSION_KHR, gl_versions[i].major,
                 EGL_CONTEXT_MINOR_VERSION_KHR, gl_versions[i].minor,
                 EGL_CONTEXT_OPENGL_PROFILE_MASK_KHR, EGL_CONTEXT_OPENGL_CORE_PROFILE_BIT_KHR,
+#ifdef ENABLE_DEBUG
```
Comment (mbouron-gpsw) [inline libnodegl/glcontext_egl.c:321]: Please use DEBUG_GL instead of ENABLE_DEBUG, it is already defined when DEBUG=yes is passed to make.

**Comment by mbouron-gpsw** [comment]
I updated the branch with the following changes:
* bumped gles required to 320 for NGLI_FEATURE_GL_KHR_DEBUG,
* moved declaration of NGLI_GL_APIENTRY to glincludes
* declared GLDEBUGPROC if needed in glincludes
* prefixed declaration of gl_debug_message_callback by NGLI_GL_APIENTRY

**Comment by jmoguillansky-gpsw** [comment]
great, thanks

**Comment by mbouron-gpsw** [comment]
I updated the branch with the following changes:
* added commit: glcontext_wgl: create a debug context if DEBUG_GL is defined
* splitted commit "debug: enable gl debug context", into glcontext_egl: create a debug context if DEBUG_GL is defined and gctx_gl: enable debug output if DEBUG_GL is defined
* renamed gl_khr_debug to khr_debug and defined compatibility defines under NGL_KHR_DEBUG_COMPAT_INCLUDES so all the relevant defines are available on all platform (fixes build on Android/macOS with DEBUG_GL)

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**